### ISSUE TITLE
storage resource lifecycle management is scattered everywhere

### ISSUE THREAD

**Issue by maplain** [vmware/vic#4830]
#For stories, please include the information below:

**Details:**
when create a vch, conceptually there would be these parts for now afaic: VIC, kvStore, folder for applianceVM, and volume_store/volumes(volumes is created later under volume_store/). Their creations are scattered everywhere: 
1. volume_store is always created by createVolumeStore on vc, but by createPool on esx if volume_store has the same name as vch, since during createPool, vc will create a vApp which doesn't create a directory by default, but esx creates a resourcePool which has  a corresponding directory in datastore;
2. VIC, kvStore is created when initializing portlayer rest server handlers;
3. volumes is created at the same time as above two directories, but with a little bit different logic which depends on the fact that createVolumeStore is executed, thus url.Path will be set to [datastorexx] /xxx/xx from /xx/xx(this is another issue: https://github.com/vmware/vic/issues/1301, check code: https://github.com/vmware/vic/blob/master/lib/install/management/store_files.go#L282);
4. directory for applianceVM is created by createAppliance on vc, and by createPool on esx;

And vic-machine-delete just blindly deletes everything under directory of applianceVM which generally includes: VIC, kvSotre, and volumes if volume_store is set to be the same as image_store. This gives rise to issues we currently see: https://github.com/vmware/vic/issues/4794, https://github.com/vmware/vic/issues/4478. 

VIC, kvStore are parts of portlayer which is part of appliance, so above behavior should be fine, but It would be better to collect the creation and destruction logic at the same place to make storage resource management clearer and more transparent. at least, in my personal opinion, it may be better to have a global convention.

**Comment by matthewavery** [comment]
Yes it really needs to be unified under the design docs. There are a lot of moving parts and we need should definitely unify them soon for those who want to open source contribute 😄 thank you for bringing this up. I would estimate this at a 1. I a going to go ahead and fill it out since the `New Issue` column is for OS and Customer issues.
